.changeset/new-carrots-relax.md

```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+rename `--public` as `--experimental-public`
```

Comment:
Perhaps be more specific that we are changing this option for the `dev` command?

Reply:
It's for both dev and publish, which is the only places they're used.

Reply:
Ah! I misunderstood the commit message. I thought you were aligning the `--public` option for `dev` with the same option in `pages`. But I see now that there is no such option for `pages`.